MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,344 @@
+
+```ts
+await client.loginWithRedirect({
+  async onRedirect(url) {
+    // Redirect using Capacitor's Browser plugin
+    await Browser.open({ url });
+  }
+});
+```
+
+The above snippet aligns more with the intent, using our SDK to login but relying on Capacitor (or any other external browser) to do the actual redirect.
+
+### `buildLogoutUrl` has been removed
+
+In v1, we introduced `buildLogoutUrl` for applications that are unable to use `window.location.assign` when logging out from Auth0, a typical example is for people using v1 of our SDK with Ionic:
+
+```ts
+// Ask auth0-spa-js to build the logout URL
+const url = await client.buildLogoutUrl();
+// Redirect using Capacitor's Browser plugin
+await Browser.open({ url });
+```
+
+With v2, `buildLogoutUrl` has been removed and you should update any code that is not able to rely on `window.location.assign` to use `onRedirect` when calling `logout`:
+
+```ts
+client.logout({
+  onRedirect(url) {
+    // Redirect using Capacitor's Browser plugin
+    await Browser.open({ url });
+  }
+});
+```
+
+This method was removed because, when using our SDK, the logout method is expected to be called regardless of the browser used. Instead of calling both `logout` and `buildLogoutUrl`, you can now change the redirect behaviour when calling `logout`.
+
+### `redirectMode` has been removed from `loginWithRedirect`
+
+In v1, Auth0Client’s `loginWithRedirect` takes a `redirectMode` that can be set to any of `assign` and `replace`, allowing the users to control whether the SDK should redirect using `window.location.assign` or `window.location.replace`.
+
+```ts
+await client.loginWithRedirect({
+  redirectMode: 'replace'
+});
+```
+
+With the release of v2, we have removed `redirectMode`. If you want to use anything but `window.location.assign` to handle the redirect to Auth0, you should implement `onRedirect`:
+
+```ts
+await client.loginWithRedirect({
+  onRedirect(url) {
+    window.location.replace(url);
+  }
+});
+```
+
+### `ignoreCache` on `getTokenSilentlyhas` been replaced by `cacheMode`
+
+In v1, users can bypass the cache when calling getTokenSilently by passing ignoreCache: true.
+
+```ts
+const token = await client.getTokenSilently({ ignoreCache: true });
+```
+
+With v2, we wanted to add the ability to only retrieve a token from the cache, without contacting Auth0 if no token was found. To do so, we have removed the `ignoreCache` property and replaced it with `cacheMode` that can take any of the following three values:
+
+- **on** (default): read from the cache caching, but fall back to Auth0 as needed
+- **off**: ignore the cache, instead always call Auth0
+- **cache-only**: read from the cache, don’t fall back to Auth0
+
+Any code that was previously using `ignoreCache: true` should be changed to use `cacheMode: 'off'`:
+
+```ts
+const token = await client.getTokenSilently({ cacheMode: 'off' });
+```
+
+### `application/x-www-form-urlencoded` is used by default instead of `application/json`
```

Comment:
Might be worth mentioning in here somewhere that this provides a small performance benefit.